plugins/repository-azure/src/main/java/org/opensearch/repositories/azure/AzureStorageSettings.java

```diff
@@ -45,13 +51,16 @@
 import org.opensearch.core.common.settings.SecureString;
 
 import java.net.InetAddress;
+import java.net.URI;
 import java.net.UnknownHostException;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.Locale;
 import java.util.Map;
+import java.util.function.Function;
 
 final class AzureStorageSettings {
+    private final ClientLogger logger = new ClientLogger(AzureStorageSettings.class);
```

Comment:
Please use the one from `AzureStorageService` (it could be passed as an argument to `getStorageEndpoint`